src/checkers/inference/model/InequalityConstraint.java

```diff
@@ -27,6 +27,11 @@ protected static Constraint create(Slot first, Slot second, AnnotationLocation l
                     + first + " Supertype: " + second);
         }
 
+        // Normalization cases:
+        // C1 != C2 => TRUE/FALSE depending on annotation
+        // V != V => FALSE
+        // otherwise => CREATE_REAL_INEQUALITY_CONSTRAINT
+
         if (first instanceof ConstantSlot && second instanceof ConstantSlot) {
```

Comment:
slot manager should already ensure that there is never the same annotation mirror mapped to two different constant IDs...

so by ID equality this should already be normalizeable, without looking through AnnotationUtils

Reply:
Keep constant - constant case for creating both TRUE and FALSE constraints, but use ID equality (or just ref equal)